Fix pg-copy-stream to work with node-postgres 4.x and node 4.2.2

chriskinsman opened the pull request:
- Eliminate unwiring/rewiring of listeners.  Not possible to get right due to .on() for some and .once() for others.  Reattaching listeners in some node versions switches flowing mode and loses data.  Instead just fork stream and send it to pg-copy-stream and the existing pg.connection.stream.  Ending pg-copy-stream is just an unpipe() call
- Upgade devDependency to pg v4.4.3
- Change test references from pg.js to pg
- Remove unneeded listener test due to change in strategy
- Require Postgres 9.2 for Travis tests

brianc commented:
I super appreciate this. The tests are failing for node@0.10.x - should we just drop support for that version of node? It's several years old now.

chriskinsman commented:
I think with this change it should work in node@0.10.x.

This does have some additional overhead since it fires handleCopyData into the query object defined by copy-to.js.  I thought about removing the listener for copyData in submit since it just rewired up when readyForQuery is received.  Have to test that some more though

chriskinsman commented:
I was wrong.  Looks like there is an issue connecting to/from in the binary test in node@0.10.x.  I updated travis to use node@0.12.x and dropped node@0.10.x

chriskinsman commented:
This should remove the need for: https://github.com/brianc/node-pg-copy-streams/pull/28

agolf88 commented:
I verified this is working on Node 4.2.2 with PG 4.4.4.  Thanks chriskinsman!

cbaron commented:
I will confirm that this fork fixed my copy stream issues!  My image server is humming.

brianc commented:
@cbaron are you confirming now? If so that's great news & I can integrate this & push a new version out pronto!

cbaron commented:
I did not do a code review, or run tests, I simply used this fork and was able to stream data out of multiple connections at once without issues.

brianc commented:
Okay great! I'll do the rest! 👍  thanks for your help @chriskinsman and sorry for the delay!

cbaron commented:
Hm, there may still be an issue with releasing the connection.

cbaron commented:
Hm, now it seems my non copy-stream query connections are not being released back to the pool.

brianc commented:
could you reproduce it in a test case or code snippet?

On Tue, May 3, 2016 at 1:32 PM, cbaron <notifications@github.com> wrote:

> Hm, now it seems my non copy-stream query connections are not being
> released back to the pool.
>
> —
> You are receiving this because you modified the open/close state.
> Reply to this email directly or view it on GitHub
> <https://github.com/brianc/node-pg-copy-streams/pull/35#issuecomment-216622766>
>

cbaron commented:
In `https://github.com/cbaron/patchwork/blob/dev/dal/postgres.js`, a module that I abstract the DAL for the application I'm working on.

When I `console.log(typeof this.done())` I'm getting undefined. -- Investigating further now..

cbaron commented:
Never mind, I think this is on me.

Thanks again.

brianc commented:
no worries! Happy to help!

On Tue, May 3, 2016 at 1:52 PM, cbaron <notifications@github.com> wrote:

> Never mind, I think this is on me.
>
> Thanks again.
>
> —
> You are receiving this because you modified the open/close state.
> Reply to this email directly or view it on GitHub
> <https://github.com/brianc/node-pg-copy-streams/pull/35#issuecomment-216628477>
>

cbaron commented:
@brianc, Unfortunately, I believe there is still a lingering issue regarding this pull request.

https://gist.github.com/cbaron/963311847387b96925cb628a62907c82

Run this code, the last query will not occur until.

If you make a change so that only 9 copy-streams are made, all queries are run.  Can you confirm ?
